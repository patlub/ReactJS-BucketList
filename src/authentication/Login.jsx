// Styles
import '../App.css';

import React, {Component} from 'react';
import {Link, Redirect} from 'react-router-dom';

// configs
import {baseURL} from '../configs/config';

// Third party
import axios from 'axios'
import {NotificationContainer, NotificationManager} from 'react-notifications';

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn: localStorage.getItem('token'),
            email: '',
            password: '',
        };

    }

    /*
    * Fired when input changes
    * @param (event) event when input changes
    * */
    onInputChanged = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
    };

    /**
     * This method handles user login
     * @param {event} event event triggering login action.
     */
    onLoginHandler = (event) => {
        event.preventDefault();
        axios.post(`${baseURL}/auth/login`,
            {
                email: this.state.email,
                password: this.state.password
            })
            .then((response) => {
                localStorage.setItem('token', response.data.token);
                this.setState({loggedIn: response.data.token});
            })
            .catch(() => {
                NotificationManager.error(`Invalid username or password`);
            });

    };

    render() {
        if (this.state.loggedIn) {
            return <Redirect to="/buckets"/>
        }
        else {
            return (
                <div className="container-fluid text-center ">
                    <div className="col-md-4 col-md-offset-4 col-sm-9 auth-box">
                        <div className="panel panel-primary">
                            <div className="panel-heading">
                                <h4 className="panel-title">Login</h4>
                            </div>
                            <div className="panel-body">
                                <form onSubmit={this.onLoginHandler}>
                                    <div className="modal-body">
                                        <div className="form-group">
                                            <input
                                                className="form-control"
                                                type="email"
                                                placeholder="Email"
                                                required
                                                name="email"
                                                value={this.state.email}
                                                onChange={this.onInputChanged}
                                            />
                                        </div>
                                        <div className="form-group">
                                            <input
                                                className="form-control"
                                                type="password"
                                                placeholder="Password"
                                                name="password"
                                                value={this.state.password}
                                                onChange={this.onInputChanged}
                                                required
                                            />
                                        </div>

                                    </div>
                                    <div className="modal-footer">
                                        <div>
                                            <button
                                                type="submit"
                                                className="btn btn-primary btn-lg btn-block">Login
                                            </button>
                                        </div>
                                        <div>
                                            <Link to="register" className="btn btn-link">Register</Link>
                                        </div>
                                    </div>
                                </form>

                            </div>
                        </div>
                    </div>
                    <NotificationContainer/>
                </div>
            );
        }
    }
}

export default Login;
