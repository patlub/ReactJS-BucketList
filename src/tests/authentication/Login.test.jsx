import React from 'react';
import { mount, shallow } from 'enzyme';
import ReactTestUtils from 'react-dom/test-utils'; // ES6
import { StaticRouter } from 'react-router';
import Login from '../../authentication/Login';
import renderer from 'react-test-renderer';


global.localStorage = {
  getItem: () => {},
};

describe('Component: Login', () => {
  it('Displays Login', () => {
    const rendered = renderer.create(
      <StaticRouter location="login" context={''}>
        <Login />
      </StaticRouter>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('updates state on input change', () => {
    const routerComponent = mount(
      <StaticRouter location="login" context={''}>
        <Login />
      </StaticRouter>,
    );
    const loginComponent = routerComponent.find('Login');
    const emailInput = loginComponent.find('[type="email"]');
    const passwordInput = loginComponent.find('[type="password"]');

    ReactTestUtils.Simulate.change(emailInput.getDOMNode(), { target: { name: 'email', value: 'patrick@gmail.com' } });
    ReactTestUtils.Simulate.change(passwordInput.getDOMNode(), { target: { name: 'password', value: 'patrick' } });
    expect(routerComponent.find('Login').nodes[0].state.email).toEqual('patrick@gmail.com');
    expect(routerComponent.find('Login').nodes[0].state.password).toEqual('patrick');
  });
});

