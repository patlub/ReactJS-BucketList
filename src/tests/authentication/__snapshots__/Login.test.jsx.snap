// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Login Displays Login 1`] = `
<div
  className="container-fluid text-center "
>
  <div
    className="col-md-4 col-md-offset-4 col-sm-9 auth-box"
  >
    <div
      className="panel panel-primary"
    >
      <div
        className="panel-heading"
      >
        <h4
          className="panel-title"
        >
          Login
        </h4>
      </div>
      <div
        className="panel-body"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="modal-body"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <div>
              <button
                className="btn btn-primary btn-lg btn-block"
                type="submit"
              >
                Login
              </button>
            </div>
            <div>
              <a
                className="btn btn-link"
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
              <a
                className="btn btn-link"
                href="/forgot_password"
                onClick={[Function]}
              >
                Forgot Password
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Displays NavBar 1`] = `
<div
  className="container-fluid text-center "
>
  <div
    className="col-md-4 col-md-offset-4 col-sm-9 auth-box"
  >
    <div
      className="panel panel-primary"
    >
      <div
        className="panel-heading"
      >
        <h4
          className="panel-title"
        >
          Login
        </h4>
      </div>
      <div
        className="panel-body"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="modal-body"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <div>
              <button
                className="btn btn-primary btn-lg btn-block"
                type="submit"
              >
                Login
              </button>
            </div>
            <div>
              <a
                className="btn btn-link"
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
              <a
                className="btn btn-link"
                href="/forgot_password"
                onClick={[Function]}
              >
                Forgot Password
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
